app/src/main/kotlin/com/hippo/ehviewer/ui/screen/DownloadsScreen.kt

```diff
@@ -253,46 +298,49 @@ fun DownloadsScreen(navigator: DestinationsNavigator) = composing(navigator) {
-                        snapshotFlow { dismissState.currentValue }.collect {
-                            if (it == SwipeToDismissBoxValue.EndToStart) {
-                                runCatching {
-                                    awaitPermissionOrCancel(confirmText = R.string.delete) {
-                                        Text(text = stringResource(R.string.delete_label, item))
-                                    }
-                                }.onSuccess {
-                                    DownloadManager.deleteLabel(item)
-                                    when (filterState.label) {
-                                        item -> switchLabel("")
-                                        null -> invalidateKey = !invalidateKey
+
+                itemsIndexed(groupLabelList, key = { _, item -> item.id }) { index, (id, item) ->
+                    val dismissState = if (DownloadsFilterMode.ARTIST == filterMode) null else {
+                        // Not using rememberSwipeToDismissBoxState to prevent LazyColumn from reusing it
+                        val dismissState = remember { SwipeToDismissBoxState(SwipeToDismissBoxValue.Settled, density, positionalThreshold = positionalThreshold,) }
+                        LaunchedEffect(dismissState) {
+                            snapshotFlow { dismissState.currentValue }.collect {
+                                if (it == SwipeToDismissBoxValue.EndToStart) {
+                                    runCatching {
+                                        awaitPermissionOrCancel(confirmText = R.string.delete) {
+                                            Text(text = stringResource(R.string.delete_label, item))
+                                        }
+                                    }.onSuccess {
+                                        DownloadManager.deleteLabel(item)
+                                        when (filterState.label) {
+                                            item -> switchLabel("")
+                                            null -> invalidateKey = !invalidateKey
+                                        }
+                                    }.onFailure {
+                                        dismissState.reset()
                                     }
-                                }.onFailure {
-                                    dismissState.reset()
                                 }
                             }
                         }
+                        dismissState
                     }
-                    ReorderableItem(reorderableLabelState, key = id!!) { isDragging ->
-                        SwipeToDismissBox2(
-                            state = dismissState,
-                            backgroundContent = {},
-                        ) {
+                    ReorderableItem(reorderableLabelState, key = id) { isDragging ->
+                        val content: @Composable () -> Unit = @Composable {
```

Comment:
You may want `movableContentOf`

Reply:
I don't think it's necessary?
Can just disable swipe & reorder.